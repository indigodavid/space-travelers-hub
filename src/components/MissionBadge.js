import React from 'react';
import PropTypes from 'prop-types';

const MissionBadge = (props) => {
  const { reserved } = props;
  if (reserved) {
    return (
      <span className="membership active">Active Member</span>
    );
  }
  return (
    <span className="membership">Not a Member</span>
  );
};

MissionBadge.propTypes = {
  reserved: PropTypes.bool.isRequired,
};

export default MissionBadge;
