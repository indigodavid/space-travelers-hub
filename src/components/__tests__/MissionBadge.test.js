import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionBadge from '../MissionBadge';

const renderMissionBadgeComponent = (reserved) => {
  const { container } = render(
    <Provider store={store}>
      <MissionBadge reserved={reserved} />
    </Provider>,
  );
  return container;
};

describe('Test Missions component', () => {
  test('Match Snapshot', () => {
    expect(renderMissionBadgeComponent(true).firstChild).toMatchSnapshot();
  });

  test('Check status when reserved is true', () => {
    expect(renderMissionBadgeComponent(true).querySelector('.active')).toBeInTheDocument();
  });

  test('Check status when reserved is true', () => {
    renderMissionBadgeComponent(false);
    expect(screen.getByText(/Not a Member/i)).toBeInTheDocument();
  });
});
