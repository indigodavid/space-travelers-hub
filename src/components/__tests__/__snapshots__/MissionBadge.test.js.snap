// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions component Match Snapshot 1`] = `
<span
  class="membership active"
>
  Active Member
</span>
`;
